### PR TITLE
Add required new option type for constrained option input types

### PR DESCRIPTION
**Pull request by jackgerrits**
Adds `int64_t`, `std::vector<uint64_t>`, `std::vector<int64_t>`

### CODE REVIEW

File: vowpalwabbit/options_types.h [hunk 1]
```diff
@@ -30,9 +30,9 @@ struct typelist
 
 // is_scalar detection relies on only accepting std::vector<T> for vector types. If that changes, update
 // is_scalar_impl below.
-using supported_options_types =
-    typelist<unsigned int, int, size_t, uint64_t, float, double, char, std::string, bool, std::vector<int>,
-        std::vector<size_t>, std::vector<float>, std::vector<double>, std::vector<char>, std::vector<std::string>>;
+using supported_options_types = typelist<unsigned int, int, size_t, uint64_t, int64_t, float, double, char, std::string,
+    bool, std::vector<int>, std::vector<uint64_t>, std::vector<int64_t>, std::vector<size_t>, std::vector<float>,
```
Comment (olgavrou) [inline vowpalwabbit/options_types.h:34]: do we also want `std::vector<unsigned int>`?

**Review by olgavrou — APPROVED** [review]

File: vowpalwabbit/options_types.h (hunk 1 above)
Comment (jackgerrits) [reply vowpalwabbit/options_types.h:34]: unsigned is already covered by `std::vector<uint64_t>`. Once I've gone through and aligned usage to either `int64_t` or `uint64_t`, then the `unsigned int`, `int` and `size_t` variants will be removed.

The diff is unclear but all this does is add `int64_t`, `std::vector<uint64_t>`, `std::vector<int64_t>`

File: vowpalwabbit/options_types.h [hunk 2]
```diff
@@ -30,9 +30,9 @@ struct typelist
 
 // is_scalar detection relies on only accepting std::vector<T> for vector types. If that changes, update
 // is_scalar_impl below.
-using supported_options_types =
-    typelist<unsigned int, int, size_t, uint64_t, float, double, char, std::string, bool, std::vector<int>,
-        std::vector<size_t>, std::vector<float>, std::vector<double>, std::vector<char>, std::vector<std::string>>;
+using supported_options_types = typelist<unsigned int, int, size_t, uint64_t, int64_t, float, double, char, std::string,
```
Comment (lalo) [inline vowpalwabbit/options_types.h:33]: what is the purpose of adding these 3 new types? or why would be a better question

Comment (lalo) [reply vowpalwabbit/options_types.h:33]: oh, I just noticed the other prs that depend on this one

Comment (jackgerrits) [reply vowpalwabbit/options_types.h:33]: Yep, the plan is to constrain options to only a few essential types compared to the several duplicate types there are now

**Pull request merged**